### ISSUE TITLE
Small question regarding gradient scaling

### ISSUE THREAD

**Issue by NDKoehler** [apache/incubator-mxnet#1632]
Where in the code is the gradient rescaled with the batchsize?

**Comment by piiswrong** [comment]
in optimizers

**Comment by taoari** [comment]
Any implementation difference comparing against Caffe? 

I am finetuning the Caltech101 dataset with the pre-trained GoogleNet model (converted from Caffe). With exactly the same hyper-parameters (learning rate, wd, momentum, etc.), it seems that MXNet converges much slower that Caffe. Is there any hyper-parameters that I should scale accordingly?

Here is the log file:

* For MXNet:

| epoch | train-accuracy | valid-accuracy | time |
| --- | --- | --- | --- |
|  1 | 0.070569 | 0.497333 | 67.3 |
|  2 | 0.503607 | 0.779000 | 67.2 |
|  3 | 0.719350 | 0.836667 | 68.1 |
|  4 | 0.791148 | 0.850000 | 66.2 |
|  5 | 0.834754 | 0.847731 | 67.8 |
|  6 | 0.872520 | 0.872667 | 68.4 |
|  7 | 0.882951 | 0.863333 | 74.7 |
|  8 | 0.901463 | 0.871667 | 78.2 |
|  9 | 0.916721 | 0.880000 | 74.0 |
| 10 | 0.923934 | 0.876975 | 76.1 |

* For Caffe (122 iters = 1 epoch)

| iter | test-accuracy |
|-----|-------------------|
| 0 | 0.0040 |
| 122 | 0.8514 |
| 244 | 0.8825 |
| 366 | 0.8861 |
| 488 | 0.9072 |
| 610 | 0.9065 |
| 732 | 0.9045 |
| 854 | 0.9132 |
